### PR TITLE
Runtime Error Fallback UI

### PR DESCRIPTION
**Pull request by tdkent**
Adds a general error boundary `error.jsx` that handles unexpected runtime errors, such as server outages preventing the execution of API requests. Thrown errors "bubble up" to the error boundary which renders a fallback UI with information and a button that reloads the current view. This prevents both crashes and unhandled errors that interrupt the application.

The following routes and actions can be tested to check that unexpected errors are bubbling to `error.jsx`:

#### New Appointment view:
- Fetch request of addresses after interacting with Addresses input
- Submitting the form

#### New Appointment "Success" view
- Fetch request for most recent submitted appointment

#### My Appointments
- Fetch request of user's appointments
- Cancelling an appointment

#### Dashboard
- Fetch request for all appointments

> [!NOTE]
> The easiest way to test is to shut down your local server before attempting fetch requests or server actions. This will simulate a server outage.

https://github.com/user-attachments/assets/6ced6817-b827-4a2c-9add-e072add3b464

### CODE REVIEW

File: client/actions/form.js [hunk 1]
```diff
@@ -1,68 +1,83 @@
...
+    const data = await response.json();
 
-  if (!response.ok) {
-    return { message: data.message };
+    if (!response.ok) {
+      return { message: data.message };
+    }
+  } catch (error) {
+    throw error;
   }
```
Comment (DalySoftware) [inline client/actions/form.js:21]: This try catch does nothing, unless I'm reading it wrong

File: client/components/form/Form.jsx [hunk 2]
```diff
@@ -53,22 +54,26 @@ export default function Form({ email }) {
...
   const [errorPath, setErrorPath] = useState("");
   const [toast, setToast] = useState(false);
   const [toastMsg, setToastMsg] = useState("");
-  const [disableBtn, setDisableBtn] = useState(null);
+  const [hasSubmitError, setHasSubmitError] = useState(false);
   const [isPending, setIsPending] = useState(false);
 
   function handleCancel() {
-    redirect("/new-appointment/cancel");
+    router.push("/new-appointment/cancel");
```
Comment (DalySoftware) [inline client/components/form/Form.jsx:67]: In case anyone else is curious why this change, this [comment](https://old.reddit.com/r/nextjs/comments/17gsis3/when_would_i_use_redirect_vs_routerpush_in_next_13/k6imjwz/) seems to explain

> Redirect and router.push() are fairly similar, the main differences are: - router.push() can only be used on the client - redirect can be used on the client and server - only router.push() pushes a new entry into the history stack so on back navigation the user will return to where they previously were
> 
> Typically unless you’re on the server or do not want to push a new entry to the history stack you should always use router.push()

Comment (tdkent) [reply client/components/form/Form.jsx:67]: Yes, correct. Although `redirect` will work in a client component, it will not work in a try/catch, and when its placed outside the try/catch, it will still execute even if an error has occurred. `router.push()` is the correct option for client components and try/catch.

File: client/actions/form.js (hunk 1 above)
Comment (tdkent) [reply client/actions/form.js]: Good catch, this is redundant

**Review by DalySoftware — APPROVED** [review]
